exportlib.php

```diff
@@ -30,134 +30,57 @@
 require_once($CFG->dirroot . "/blocks/coursefeedback/lib.php");
 require_once($CFG->libdir . '/csvlib.class.php');
 
-class feedbackexport {
-    protected $course = 0;
-    protected $feedback = 0;
-    protected $filetypes = array("csv");
-    private $content = "";
-    private $format;
-
-    public function __construct($course = 0, $feedback = 0, $seperator = "\t") {
-        global $DB;
-
-        if ($DB->record_exists("course", array("id" => $course))) {
-            $this->course = $course;
-            $this->feedback = $feedback;
-        } else {
-            print_error("courseidnotfound", "error");
-            exit(0);
-        }
-    }
-
-    public function get_filetypes() {
-        return $this->filetypes;
-    }
-
-    public function init_format($format) {
-        if (in_array($format, $this->get_filetypes())) {
-            $exportformatclass = "exportformat_" . $format;
-            $this->format = new $exportformatclass();
-            return true;
-        } else {
-            return false;
-        }
-    }
-
-    public function create_file($lang) {
-        global $CFG, $DB;
-
-        if (!isset($this->format)) {
-            print_error("format not initialized", "block_coursefeedback");
-        } else {
-            $answers = block_coursefeedback_get_answers($this->course, $this->feedback);
-            $this->reset();
-            $this->content = $this->format->build($answers, $lang);
-        }
-    }
-
-    public function get_content() {
-        return $this->content;
-    }
-
-    public function reset() {
-        $this->content = "";
-    }
-}
 
 /**
- * @author Jan Eberhardt
- * Generell format class. Doesn"t contain very much so far, but should provide basics.
+ * Export feedback data for a course.
+ *
+ * @package block
+ * @subpackage coursefeedback
+ * @copyright innoCampus, TU Berlin
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
  */
-abstract class exportformat {
-    private $type = "unknown";
+class feedback_exporter {
+    protected $csvexportwriter;
 
-    public final function get_type() {
-        return $this->type;
-    }
-
-    public abstract function build($arg1);
-}
-
-/**
- * @author Jan Eberhardt
- * CSV export class
- */
-class exportformat_csv extends exportformat {
-    public $seperator;
-    public $newline;
-    public $quotes;
-
-    /**
-     * Set CSV options.
-     *
-     * TODO Choosable values.
-     */
     public function __construct() {
-        $this->type = "csv";
-        $this->seperator = ";";
-        $this->newline = "\n";
-        $this->quotes = "\"";
+        $this->csvexportwriter = new csv_export_writer();
     }
 
-    /**
-     * (non-PHPdoc)
-     * @see exportformat::build()
-     */
-    public function build($answers, $lang = null) {
+    public function create_file($courseid, $feedbackid) {
```

Comment:
Finde den Ansatz mit dem `csv_export_writer` prinzipiell gut. Ich habe die Funktion jetzt natürlich nicht getestet. Für sowas könnte man aber m.E. auch einen Behat Test schreiben, bei dem ein paar Dummy Fragen und Antworten generiert werden, dann der Export angestoßen wird und dann die resultierende CSV Datei entsprechend gecheckt wird. Falls das mit dem Export wichtig ist (wovon ich ausgehe), wäre das den einmaligen Aufwand vielleicht wert.

Reply:
Tests sind natürlich wünschenswert und wichtig, das würde ich aber am liebsten ans Ende stellen wenn die anderen Todos in der Pipe auch integriert sind.
Ich habe dafür ein [issue](https://github.com/eberhardt/moodle-block_coursefeedback/issues/33) erstellt